### PR TITLE
[Fix] Fix llama2 70b & qwen quantization error

### PR DESCRIPTION
**Pull request by pppppM**
1. [x] llama2 70b
2. [x] qwen

### CODE REVIEW

**Comment by lvhan028** [comment]
Can we upload their quantized models to huggingface?

File: lmdeploy/lite/apis/auto_awq.py [hunk 1]
```diff
@@ -26,24 +28,42 @@
 
 
 def auto_awq(model: str,
+             work_dir: str,
```
Comment (lvhan028) [inline lmdeploy/lite/apis/auto_awq.py:31]: What's the motivation of changing `work_dir` from optional argument to positional argument? I tried the following codes, it  raised an error:

```plaintext
(triton) [lvhan@SH-IDC1-10-140-24-140 llmdeploy]$ python lmdeploy/lite/apis/auto_awq.py --model /nvme/shared_data/llama2/huggingface/llama-2-7b-chat/
[2023-08-21 11:47:05,460] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
ERROR: The function received no value for the required argument: work_dir
Usage: auto_awq.py MODEL WORK_DIR <flags>
  optional flags:        --w_bits | --w_sym | --w_group_size | --device

For detailed information on this command, run:
  auto_awq.py --help
```

**Comment by LZHgrla** [comment]
Can we merge multi_scale `scales` into one_head, and thus enable the smooth for llama2-70b's `self_attn.v_proj` and `self_attn.o_proj`?

Like this
```python
@torch.no_grad()
def smooth_fc_fcs(pre_fc: torch.nn.Module,
                  fcs: List[torch.nn.Module],
                  act_scales: torch.Tensor,
                  group_size: int = -1,
                  alpha: float = 0.5) -> torch.Tensor:
    """Smooth weights of a fully connected layer and its downstream layers.

    :param pre_fc: Previous Fully Connected layer
    :param fcs: List of Fully Connected modules
    :param act_scales: Activation scales
    :param alpha: Scaling factor (default is 0.5)
    :return: Scales
    """
    device, dtype = pre_fc.weight.device, pre_fc.weight.dtype

    concat_w = torch.cat([fc.weight for fc in fcs], dim=0)
    w_scales = get_weight_scale(concat_w, group_size)

    act_scales = act_scales.to(device=device, dtype=dtype)

    scales = (act_scales.pow(alpha) /
              w_scales.pow(1 - alpha)).clamp(min=1e-4).to(device).to(dtype)
    scales = scales / (scales.max() * scales.min()).sqrt()

    size_scales = scales.size(0)
    size_pre_fc = pre_fc.weight.size(0)

    # (for llama2) use group query attention, pre_fc is v_proj, fc is o_proj
    if size_pre_fc < size_scales and size_scales % size_pre_fc == 0:
        # merge multi_head scales to one_head
        scales = scales.view(-1, size_pre_fc).mean(0)
        size_scales = scales.size(0)

    # (for qwen) pre_fc is packed QKV, only V needs to scale
    if size_pre_fc > size_scales and size_pre_fc % size_scales == 0 \
            and size_pre_fc // size_scales == 3:

        pre_fc.weight[-size_scales:].div_(scales.view(-1, 1))

        if getattr(pre_fc, 'bias', None) is not None:
            pre_fc.bias[-size_scales:].div_(scales)
    else:
        pre_fc.weight.div_(scales.view(-1, 1))

        if getattr(pre_fc, 'bias', None) is not None:
            pre_fc.bias.div_(scales)

    for fc in fcs:
        size_fc = fc.weight.size(1)
        # (for llama2) map one_head scales to multi_head
        if size_scales < size_fc and size_fc % size_scales == 0:
            cur_scales = scales.repeat(size_fc // size_scales)
        else:
            cur_scales = scales
        fc.weight.mul_(cur_scales.view(1, -1))

    for p in pre_fc.parameters():
        assert torch.isnan(p).sum() == 0
    for fc in fcs:
        for p in fc.parameters():
            assert torch.isnan(p).sum() == 0

    return scales
```

**Comment by lvhan028** [comment]
I tried to quantize Llama2-70B-chat model, but got an error when executing `auto_awq`
```plaintext
RuntimeError: [enforce fail at inline_container.cc:337] . unexpected pos 4566829760 vs 4566829656
```

The commands are:

```shell
python lmdeploy/lite/apis/calibrate.py --model /nvme/shared_data/llama2/huggingface/llama-2-70b-chat-hf/
```
```shell
python lmdeploy/lite/apis/auto_awq.py --model /nvme/shared_data/llama2/huggingface/llama-2-70b-chat-hf/ --work-dir ./work_dir
```

**Comment by lvhan028** [comment]
May update FAQ to present the reason to "RuntimeError: [enforce fail at inline_container.cc:337] . unexpected pos 4566829760 vs 4566829656"

**Review by lvhan028 — APPROVED** [review]
LGTM after updating FAQ

**Review by LZHgrla — APPROVED** [review]

**Pull request merged**